src/librustc/middle/privacy.rs

```diff
@@ -11,16 +11,31 @@ use syntax::ast::NodeId;
 // Accessibility levels, sorted in ascending order
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum AccessLevel {
-    // Superset of Reachable used to mark impl Trait items.
+    /// Superset of `AccessLevel::Reachable` used to mark impl Trait items.
     ReachableFromImplTrait,
-    // Exported items + items participating in various kinds of public interfaces,
-    // but not directly nameable. For example, if function `fn f() -> T {...}` is
-    // public, then type `T` is reachable. Its values can be obtained by other crates
-    // even if the type itself is not nameable.
+    /// Superset of `AccessLevel::Reachable` that includes `ItemKind::Use` items that
+    /// are used to transitively export an item. Used by `unreachable_pub` lint so that
+    /// these items are considered reachable.
+    ///
+    /// ie. `pub use self::bar::Bar` in the below example:
+    /// ```
+    /// mod foo {
+    ///     mod bar {
+    ///         pub struct Bar;
+    ///     }
+    ///     pub use self::bar::Bar;
+    /// }
+    /// pub use foo::Bar;
+    /// ```
+    ReachableFromTransitiveUse,
```

Comment:
This should be equivalent to `AccessLevel::Exported` for all purposes, so you can avoid introducing it.

Reply:
Adding missing stability attributes would be preferable, I think.
Even if stability attributes are not checked on imports right now, requiring them is still the correct intended behavior.

Reply:
Changed this.